ext/neural_snow/NeuralSnow.jl

```diff
@@ -0,0 +1,223 @@
+Helper function for evaluating the neural network in a pointwise manner over a `ClimaCore.Field`
+and returning the output in a broadcastable way.
+"""
+function eval_nn(
+    model,
+    z::FT,
+    swe::FT,
+    P::FT,
+    T::FT,
+    R::FT,
+    qrel::FT,
+    u::FT,
+)::FT where {FT}
+    return model([z, swe, qrel, R, u, T, P])[1]
+end
+
+"""
+    dzdt(density::NeuralDepthModel, model::SnowModel{FT}, Y, p, t) where {FT}
+Returns the change in snow depth (rate) given the current model state and the `NeuralDepthModel`
+density paramterization, passing the approximate average of the forcings over the last 24 hours instead of
+the instantaneous value.
+"""
+function dzdt(density::NeuralDepthModel, Y)
+    return eval_nn.(
+        Ref(density.z_model),
+        Y.snow.Z,
+        Y.snow.S,
+        Y.snow.P_avg,
+        Y.snow.T_avg,
+        Y.snow.R_avg,
+        Y.snow.Qrel_avg,
+        Y.snow.u_avg,
+    )
+end
+
+"""
+    clip_dZdt(S::FT, Z::FT, dSdt::FT, dZdt::FT, Δt::FT)::FT
+A helper function which clips the tendency of Z such that
+its behavior is consistent with that of S.
+"""
+function clip_dZdt(S::FT, Z::FT, dSdt::FT, dZdt::FT, Δt::FT)::FT where {FT}
+    if (S + dSdt * Δt) <= eps(FT)
+        return -Z / Δt
+    elseif (Z + dZdt * Δt) < (S + dSdt * Δt)
+        return (S - Z) / Δt + dSdt
+    else
+        return dZdt
+    end
+end
+
+"""
+    update_density_prog!(density::NeuralDepthModel, model::SnowModel, Y, p)
+Updates all prognostic variables associated with density/depth given the current model state and the `NeuralDepthModel`
+density paramterization.
+"""
+function update_density_prog!(
+    density::NeuralDepthModel,
+    model::SnowModel,
+    dY,
+    Y,
+    p,
+)
+
+    dY.snow.Z .=
+        clip_dZdt.(
+            Y.snow.S,
+            Y.snow.Z,
+            dY.snow.S, #assumes dY.snow.S is updated (and clipped) before dY.snow.Z
+            dzdt(density, Y),
+            model.parameters.Δt,
+        )
+
+    @. dY.snow.P_avg =
+        density.α / model.parameters.Δt * (abs(p.drivers.P_snow) - Y.snow.P_avg)
+    @. dY.snow.T_avg =
+        density.α / model.parameters.Δt *
+        (p.drivers.T - model.parameters.earth_param_set.T_freeze - Y.snow.T_avg)
+    @. dY.snow.R_avg =
+        density.α / model.parameters.Δt * (p.drivers.SW_d - Y.snow.R_avg)
+    dY.snow.Qrel_avg .=
+        density.α / model.parameters.Δt .* (
+            Thermodynamics.relative_humidity.(
+                model.boundary_conditions.atmos.thermo_params,
+                p.drivers.thermal_state,
+            ) .- Y.snow.Qrel_avg
+        )
+    @. dY.snow.u_avg =
+        density.α / model.parameters.Δt * (p.drivers.u - Y.snow.u_avg)
+end
+
+end
+
+#Todo: Write tests for these functions
+#Todo: address Anderson1976 sensitivity for small precision snowpacks (z = 0 when SWE > 0 or z > 0 when SWE = 0, but on the order of 1e-10 < eps(Float32), doesn't make NaN at present though)
```

Comment:
I propose we move Anderson1976 from this PR if you do not feel up to writing tests for it and want to merge something closer to what you have already. And then you can immediately open another PR adding in the Anderson parameterization code (with unit tests, the correct units, etc)